### DISCUSSION TITLE
multiple GPIO DS18x20

### DISCUSSION THREAD

**Discussion by md5sum-as** [arendst/Tasmota#16811] [General]
I have an idea to implement support for multiple DS18x20 GPIO. Such functionality will allow, for example, to divide a long line with a large electrical capacity into several short lines. I think 4 pairs of DS18x20/DS18x20_o will be enough. I suggest enabling this mode using a separate compiler flag, for example DS18x20_MULTI_GPIOs in my_user_config or user_config_override file.
Any comments?

**Comment by lalo-uy** [comment]
I will love to see that working and count me for testing.

**Comment by Romik4x4** [comment]
Excellent. Much needed functionality. I often need to measure the temperature on the board itself and connect external temperature sensors. DS18B20 sensors are quite cheap, so we use them. I will be glad to help in testing as the project is assembled and ready for tests.

**Comment by md5sum-as** [comment]
Hello.
The function is ready for testing in [repo](https://github.com/md5sum-as/Tasmota). So far only on esp8266. 
Setup example:
![image](https://user-images.githubusercontent.com/13337275/195978820-c7be2f6a-34c4-461f-af1b-0770d65842ef.png)
Example of the result:
![image](https://user-images.githubusercontent.com/13337275/195978878-2f37c9bd-72c6-4ec9-b44d-9e3e668e39a0.png)

I can't test "Shelly dual pin mode" and "parasite power". It will be good if you help me.

**Reply by barbudor** [reply]
Do you support multiple DS18x20 on multiple GPIO ?
If so, you would need a double index scheme like `DS18B20-1-1`

**Reply by md5sum-as** [reply]
No. Naming remains end-to-end. If it is necessary to distinguish between sensors, then use aliases
![image](https://user-images.githubusercontent.com/13337275/195985274-fc862780-416d-4396-b9db-51adeabbbb66.png)

**Comment by md5sum-as** [comment]
ESP32 ready for test
![image](https://user-images.githubusercontent.com/13337275/195996137-1de1e58b-547d-430c-a248-984a60a6ad00.png)
![image](https://user-images.githubusercontent.com/13337275/195996155-fa00e00c-ad97-4104-98f8-0e6495de7e99.png)

**Comment by Romik4x4** [comment]
I checked everything. Everything works. Great job.
![ssrv2-TWO-ds18b20](https://user-images.githubusercontent.com/7497569/196366430-cecae79d-8d0c-4074-b5c7-48e8ad8e7538.png)
![twods18b20](https://user-images.githubusercontent.com/7497569/196366432-cdd78f3f-3e68-41b6-a398-f37a6b3b8814.png)

**Comment by VASTechAD** [comment]
Hello! 
Tested myself and works great!
One idea! It will be nice that instead of word "Temperature" to show ID address of that sensor. I think that anyone working with this temperature sensors knows that express temperature.
Keep it up!

**Comment by lalo-uy** [comment]
In the web UI or the json? Please do not change the json.

**Comment by VASTechAD** [comment]
Web UI
